pkg/sources/docker/docker.go

```diff
@@ -139,6 +119,22 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ .
 	workers := new(errgroup.Group)
 	workers.SetLimit(s.concurrency)
 
+	// if namespace is set and no images are specified, fetch all images in that namespace.
+	registryNamespace := s.conn.GetNamespace()
+	if registryNamespace != "" && len(s.conn.Images) == 0 {
+		start := time.Now()
+		namespaceImages, err := GetNamespaceImages(ctx, registryNamespace, s.conn.GetRegistryToken())
+		if err != nil {
+			ctx.Logger().Error(err, "failed to list namespace images", "namespace", registryNamespace)
+
+			return nil
+		}
```

Comment:
I think we should just do a `fmt.Errorf` to wrap and then return the error; this should make the scanner report unhealthy